crates/ruff_db/src/system/test.rs

```diff
@@ -71,6 +73,30 @@ impl System for TestSystem {
         }
     }
 
+    fn virtual_path_metadata(&self, path: &SystemVirtualPath) -> Result<Metadata> {
+        match &self.inner {
+            TestSystemInner::Stub(fs) => fs.virtual_path_metadata(path),
```

Comment:
I think I would add a hash map storing the virtual files to system instead of adding it to `MemoryFileSystem` because `MemoryFileSystem` isn't a full `System`, it's just a building block for `System` implementations that don't want to use a real fs.

Reply:
I don't think I follow this. Would this be storing a hashmap (key is virtual file, value is system) or just storing a hashmap from virtual file path to virtual file in the `TestSystem`?

Reply:
Thinking about this more, let's just add it to `MemoryFileSystem` that internally stores a ` FxHashMap<SystemVirtualPathBuf, File>` (or `DashMap`)